test/integration/resourcemanager/managedresource/resource_test.go

```diff
@@ -245,22 +248,56 @@ var _ = Describe("ManagedResource controller tests", func() {
 			})
 		})
 
+		Describe("resource data changes", func() {
```

Comment:
I still don't see a test case in this section that uses multiple secret keys. Can you please enhance the test case to use multiple secret keys and assert the status field is set correctly before and after changing the secret data?

Reply:
Is [this test ](https://github.com/gardener/gardener/blob/e8763a5bb28def93443fd701688826c8cafc282c/test/integration/resourcemanager/managedresource/resource_test.go#L302) case fine ?